Makefile

```diff
@@ -0,0 +1,101 @@
+GOCMD=go
+GOTEST=$(GOCMD) test
+GOVET=$(GOCMD) vet
+BINARY_NAME=Spinup
+VERSION?=0.0.0
+SERVICE_PORT?=3000
+DOCKER_REGISTRY?= #if set it should finished by /
+EXPORT_RESULT?=false # for CI please set EXPORT_RESULT to true
+
+GREEN  := $(shell tput -Txterm setaf 2)
+YELLOW := $(shell tput -Txterm setaf 3)
+WHITE  := $(shell tput -Txterm setaf 7)
+CYAN   := $(shell tput -Txterm setaf 6)
+RESET  := $(shell tput -Txterm sgr0)
+
+.PHONY: all test build
+
+all: help
+
+## Build:
+build: ## Build your project and put the output binary in out/bin/
+	mkdir -p out/bin
+	GO111MODULE=on $(GOCMD) build -o out/bin/$(BINARY_NAME) .
+
+clean: ## Remove build related file 
+	rm -fr ./bin
+	rm -fr ./out
+	rm -f ./junit-report.xml checkstyle-report.xml ./coverage.xml ./profile.cov yamllint-checkstyle.xml
+
+
+watch: ## Start the Spinup -. FULL in watch model
+	air
+
+watch-install: ## Install Watch Tools
+	go install github.com/cosmtrek/air@master
+## Test:
+test: ## Run the tests of the project
+ifeq ($(EXPORT_RESULT), true)
+	GO111MODULE=off go get -u github.com/jstemmer/go-junit-report
+	$(eval OUTPUT_OPTIONS = | tee /dev/tty | go-junit-report -set-exit-code > junit-report.xml)
+endif
+	$(GOTEST) -v -race ./... $(OUTPUT_OPTIONS)
+
+coverage: ## Run the tests of the project and export the coverage
+	$(GOTEST) -cover -covermode=count -coverprofile=profile.cov ./...
+	$(GOCMD) tool cover -func profile.cov
+ifeq ($(EXPORT_RESULT), true)
+	GO111MODULE=off go get -u github.com/AlekSi/gocov-xml
+	GO111MODULE=off go get -u github.com/axw/gocov/gocov
+	gocov convert profile.cov | gocov-xml > coverage.xml
+endif
+
+## Dependencies:
+dep:  ## download your dependencies
+	go mod download
```

Comment:
just some thoughts...
can we have `dep` do the following:
- download go mod dependencies
- install dev dependecies (i.e install `golangci-lint` and maybe `air`) if not available
- update the user's PATH variable so that the dev dependencies are available?

I think it will simplify the  `make` commands instead of having multiple separate commands for installing air and linters